File: spec/functional_test/fixtures/ruby/sinatra/test/app_test.rb
# Regression guard: Minitest's `*_test.rb` convention skips file
# scanning. Routes registered in this file (the inline TestApp
# Sinatra::Base subclass below) should NOT surface as endpoints.
require 'sinatra/base'
require 'minitest/autorun'

class TestApp < Sinatra::Base
  get '/should-not-appear-test-get' do
    'ok'
  end

  post '/should-not-appear-test-post' do
    'ok'
  end
end

class AppTest < Minitest::Test
  def test_noop
    assert true
  end
end

File: spec/functional_test/fixtures/javascript/hono/app.test.ts
// Regression guard: hono's own `*.test.ts` suites register routes
// via `app.on('GET', '/path', ...)` to exercise the typing layer.
// The primary route extractor already skips files that match a
// test-stub marker; the auxiliary `extract_on_routes` pass in the
// Hono analyzer has to honor the same gate, or these routes leak
// out as if they were real endpoints. None of the URLs below should
// surface in the fixture's expected-endpoints list.
import { Hono } from 'hono'

const app = new Hono()

app.on('GET', '/should-not-appear-1', (c) => c.json({}))
app.on('GET', '/should-not-appear-2', (c) => c.json({}))
app.on('POST', '/should-not-appear-3', (c) => c.json({}))
